### PR TITLE
wrap default openapi error handler

### PR DESCRIPTION
**Pull request by isinyaaa**
## Description
Resolves: #95

## How Has This Been Tested?
Tested with making POST/PATCH API calls with random missing/mispelled fields.

## Merge criteria:

- [x] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: internal/server/openapi/helpers.go [hunk 1]
```diff
@@ -21,6 +23,13 @@ func Response(code int, body interface{}) ImplResponse {
 	}
 }
 
+func Error(code int, err error) ImplResponse {
```
Comment (tarilabs) [inline internal/server/openapi/helpers.go:26]: to avoid confusing this, should we name this `ErrorResponse` ? wdyt?

**Review by tarilabs — COMMENTED** [review]
Thank you @isinyaaa !

I believe you must include:
- `internal/server/openapi/error.go`
- `internal/server/openapi/helpers.go`

in the file `.openapi-generator-ignore`, or otherwise next generation if the OpenAPI file is changed would overwrite your changes.

Can you check this before we merge this PR please?

The other suggestion below is optional.

**Comment by isinyaaa** [comment]
> Thank you @isinyaaa !
> 
> I believe you must include:
> 
> * `internal/server/openapi/error.go`
> * `internal/server/openapi/helpers.go`
> 
> in the file `.openapi-generator-ignore`, or otherwise next generation if the OpenAPI file is changed would overwrite your changes.

It doesn't seem to be the case that these files are updated at all. Trying to regen manually with `FORCE_SERVER_GENERATION=1 make` yields no diff, but I can include them in the ignore list to be on the safe side.

Will update the suggestion.

**Comment by tarilabs** [comment]
> It doesn't seem to be the case that these files are updated at all. Trying to regen manually with FORCE_SERVER_GENERATION=1 make yields no diff,

You right, very strange 🤔 ; also tried on this end by removing the file, it is indeed not re-generated.
Could be it is one of the files which was only copy-pasted originally, not sure.

Thanks anyway for adding it to the ignore list, at least we have 1 source-of-truth for files which we have amended!

I just noticed now, this claims to resolve an issue but we didn't add any tests.
Do you think you can add it, please?

Since this is on the "REST surface" I believe a Robot test would be the simplest to do, but open to additional suggestions

**Review by tarilabs — APPROVED** [review]
I'm glad it's automatically picked up by https://github.com/kubeflow/model-registry/blob/main/.github/workflows/run-robot-tests.yaml#L57 without changes, thank you for having split into "negative test" file 👍 

/lgtm
/approve

**Pull request merged**